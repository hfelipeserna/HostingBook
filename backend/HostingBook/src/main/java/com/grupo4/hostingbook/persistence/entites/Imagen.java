package com.grupo4.hostingbook.persistence.entites;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "imagenes")
public class Imagen {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="imagen_id")
    private Long imagenId;
    private String imagenTitulo;
    @Column(name="imagen_URL")
    private String imagenUrl;

    public Long getImagenId() {
        return imagenId;
    }

    public void setImagenId(Long imagenId) {
        this.imagenId = imagenId;
    }

    public String getImagenTitulo() {
        return imagenTitulo;
    }

    public void setImagenTitulo(String imagenTitulo) {
        this.imagenTitulo = imagenTitulo;
    }

    public String getImagenUrl() {
        return imagenUrl;
    }

    public void setImagenUrl(String imagenUrl) {
        this.imagenUrl = imagenUrl;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Imagen imagen = (Imagen) o;
        return Objects.equals(imagenId, imagen.imagenId) && Objects.equals(imagenTitulo, imagen.imagenTitulo) && Objects.equals(imagenUrl, imagen.imagenUrl);
    }

    @Override
    public int hashCode() {
        return Objects.hash(imagenId, imagenTitulo, imagenUrl);
    }

}
