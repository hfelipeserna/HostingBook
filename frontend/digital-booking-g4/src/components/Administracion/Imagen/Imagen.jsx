import { useEffect, useState } from "react";
import DropInput from "../DropInput/DropInput";
import EstandarInput from "../EstandarInput/EstandarInput";
import styles from "./Imagen.module.css";

export default function Imagen({
    value,
    index,
    agregarImagen,
    handleAdd,
    handleDelete
}) {
    const [imagenDetails, setImagenDetails] = useState({
        url: "",
        descripcion: "",
    });

    const[ botonHabilitado, setBotonHabilitado]=useState(true);

    useEffect(() =>{
        const botonHabilitado = imagenDetails.url !== "" && imagenDetails.descripcion !== "";
        setBotonHabilitado(botonHabilitado);
    }, [imagenDetails])

    const setUrl = (url) =>
    setImagenDetails({
        url: url,
        descripcion: imagenDetails.descripcion,
    });

    const setDescripcion = (descripcion) =>
    setImagenDetails({
        url: imagenDetails.url,
        descripcion: descripcion,
    });
    
    return (
        <div className={styles.fila}>
          <EstandarInput
            setOnChangeItem={setUrl}
            name="url"
            placeholder="Insertar https://"
          />
          <DropInput
            setOnChangeItem={setDescripcion}
            name="descripcion"
            placeholder="Descripción"
          />
          <div className={styles.containerButton}>
            {index === 0 ? (
              <div className={styles.containerBotones}>
                <button
                  className={styles.add}
                  disabled={!botonHabilitado}
                  onClick={() => {
                    agregarImagen(imagenDetails);
                    setBotonHabilitado(false)
                  }}
                >
                  <i className="fas fa-check"></i>
                </button>
                <button className={styles.add} onClick={handleAdd}>
                  <i className="fas fa-plus" title="Clic aquí para agregar otra imágen"></i>
                </button>
              </div>
            ) : (
              <div className={styles.containerBotones}>
                <button
                  className={styles.add}
                  disabled={!botonHabilitado}
                  onClick={() => {
                    agregarImagen(imagenDetails);
                    setBotonHabilitado(false)
                  }}
                >
                  <i className="fas fa-check" title="Clic aquí para confirmar esta imágen"></i>
                </button>
                <button
                  className={styles.delete}
                  onClick={() => handleDelete(value, imagenDetails)}
                >
                  <i className="fas fa-times" title="Clic aquí para borrar esta imágen"></i>
                </button>
              </div>
            )}
          </div>
        </div>
    );
}