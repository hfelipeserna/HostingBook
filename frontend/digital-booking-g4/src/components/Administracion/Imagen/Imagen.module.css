.fila {
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  gap: 15px;
  background-color: rgba(190, 190, 190, 0.1);
  border-radius: 5px;
  padding: 0 15px 20px 15px;
  margin-top: 20px;
}

button {
  outline: none;
}

.add {
  font-size: 2rem;
  font-weight: 700;
  color: var(--color-5);
  cursor: pointer;
  background-color: var(--color-principal);
  border: 3px solid var(--color-principal);
  border-radius: 5px;
  filter: drop-shadow(0px 2px 4px rgba(0, 0, 0, 0.12));
  padding: 0 0.5vw;
  margin-top: 25px;
  transition: all ease-in-out 250ms;
}

.add:hover {
  background-color: var(--color-principal-opaco);
  border: 3px solid var(--color-principal-opaco);
}

.add:disabled {
  color: white;
  background-color: grey;
  border: 3px solid grey;
  opacity: 0.25;
}

.delete {
  font-size: 2rem;
  font-weight: 700;
  color: var(--color-5);
  cursor: pointer;
  background-color: var(--color-3);
  border: 3px solid var(--color-3);
  border-radius: 5px;
  filter: drop-shadow(0px 2px 4px rgba(0, 0, 0, 0.12));
  padding: 0 0.75vw;
  margin-top: 25px;
  transition: all ease-in-out 250ms;
}

.delete:hover {
  background-color: var(--color-2);
  border: 3px solid var(--color-2);
}

.containerBotones {
  display: flex;
  gap: 6px;
}

.invalidURL {
  background-color: #da5456;
  border-radius: 5px;
  display: flex;
  align-items: center;
  padding: 10px;
  color: #000;
  gap: 10px;
  margin: 10px 20px;
}

.invalidURL p {
  color: #000;
}

.errorMsjContainer p {
  color: #000;
  text-align: start;
}

@media (max-width: 480px) {
  .fila {
    display: flex;
    flex-direction: column;
    gap: 0px;
  }
  .containerButton {
    width: 100%;
  }
  .add,
  .delete {
    width: 100%;
    margin-top: 9px;
  }
}
