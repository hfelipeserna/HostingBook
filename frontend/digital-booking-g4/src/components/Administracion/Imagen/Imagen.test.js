import Imagen from "./Imagen"
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure, mount } from "enzyme";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Imagen test", function () {
    let index;

    test("Renderiza Componente Por default", () => {
        index = 0
        const renderizacionSimple = render(<Imagen index={index}></Imagen>)
        expect(renderizacionSimple).toMatchSnapshot()

    });
    test("Renderiza Componente Por index.not0", () => {
        index = 1
        const renderizacionConIndex = render(<Imagen index={index}></Imagen>)
        expect(renderizacionConIndex).toMatchSnapshot()
    });
    test("Renderiza Componente Por Url invalida", () => {
        index = 1
        const renderizacionConIndex = mount(<Imagen index={index}></Imagen>)
        const inputParaCambiarUrl = renderizacionConIndex.find(".estandarInput")
        inputParaCambiarUrl.simulate('focus');
        inputParaCambiarUrl.simulate('change', { target: { value: 'a' } })
        inputParaCambiarUrl.simulate('keyDown', {
            which: 27,
            target: {
                blur() {
                    inputParaCambiarUrl.simulate('blur');
                },
            },
        })
        expect(renderizacionConIndex).toMatchSnapshot()
    });



});

