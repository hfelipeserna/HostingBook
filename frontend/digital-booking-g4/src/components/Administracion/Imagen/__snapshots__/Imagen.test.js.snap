// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Imagen test Renderiza Componente Por Url invalida 1`] = `
<div
  className="container"
>
  <div
    className="fila"
  >
    <div
      className="containerEstandarInput"
    >
      <label
        className="labelEstandarInput"
      />
      <div
        className="sombraConAutocompletado"
      >
        <input
          className="estandarInput"
          name="url"
          onChange={[Function]}
          placeholder="Insertar https://"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="containerDropInput"
    >
      <label
        className="labelDropInput"
      />
      <div
        className="cityContainer divDrawer"
      >
        <input
          autoComplete="off"
          className="input"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Descripción"
          type="text"
        />
      </div>
    </div>
    <div
      className="containerButton"
    >
      <div
        className="containerBotones"
      >
        <button
          className="add"
          disabled={true}
          onClick={[Function]}
        >
          <i
            className="fas fa-check"
          />
        </button>
        <button
          className="delete"
          onClick={[Function]}
        >
          <i
            className="fas fa-times"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="invalidURL"
  >
    <i
      className="fas fa-exclamation-triangle"
    />
    <p>
      La URL ingresada no es una imagen
    </p>
  </div>
</div>
`;

exports[`Imagen test Renderiza Componente Por default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="fila"
        >
          <div
            class="containerEstandarInput"
          >
            <label
              class="labelEstandarInput"
            />
            <div
              class="sombraConAutocompletado"
            >
              <input
                class="estandarInput"
                name="url"
                placeholder="Insertar https://"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerDropInput"
          >
            <label
              class="labelDropInput"
            />
            <div
              class="cityContainer divDrawer"
            >
              
              <input
                autocomplete="off"
                class="input"
                placeholder="Descripción"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerButton"
          >
            <div
              class="containerBotones"
            >
              <button
                class="add"
                disabled=""
              >
                <i
                  class="fas fa-check"
                />
              </button>
              <button
                class="add"
              >
                <i
                  class="fas fa-plus"
                />
              </button>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="fila"
      >
        <div
          class="containerEstandarInput"
        >
          <label
            class="labelEstandarInput"
          />
          <div
            class="sombraConAutocompletado"
          >
            <input
              class="estandarInput"
              name="url"
              placeholder="Insertar https://"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerDropInput"
        >
          <label
            class="labelDropInput"
          />
          <div
            class="cityContainer divDrawer"
          >
            
            <input
              autocomplete="off"
              class="input"
              placeholder="Descripción"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerButton"
        >
          <div
            class="containerBotones"
          >
            <button
              class="add"
              disabled=""
            >
              <i
                class="fas fa-check"
              />
            </button>
            <button
              class="add"
            >
              <i
                class="fas fa-plus"
              />
            </button>
          </div>
        </div>
      </div>
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Imagen test Renderiza Componente Por index.not0 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="fila"
        >
          <div
            class="containerEstandarInput"
          >
            <label
              class="labelEstandarInput"
            />
            <div
              class="sombraConAutocompletado"
            >
              <input
                class="estandarInput"
                name="url"
                placeholder="Insertar https://"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerDropInput"
          >
            <label
              class="labelDropInput"
            />
            <div
              class="cityContainer divDrawer"
            >
              
              <input
                autocomplete="off"
                class="input"
                placeholder="Descripción"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="containerButton"
          >
            <div
              class="containerBotones"
            >
              <button
                class="add"
                disabled=""
              >
                <i
                  class="fas fa-check"
                />
              </button>
              <button
                class="delete"
              >
                <i
                  class="fas fa-times"
                />
              </button>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="fila"
      >
        <div
          class="containerEstandarInput"
        >
          <label
            class="labelEstandarInput"
          />
          <div
            class="sombraConAutocompletado"
          >
            <input
              class="estandarInput"
              name="url"
              placeholder="Insertar https://"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerDropInput"
        >
          <label
            class="labelDropInput"
          />
          <div
            class="cityContainer divDrawer"
          >
            
            <input
              autocomplete="off"
              class="input"
              placeholder="Descripción"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="containerButton"
        >
          <div
            class="containerBotones"
          >
            <button
              class="add"
              disabled=""
            >
              <i
                class="fas fa-check"
              />
            </button>
            <button
              class="delete"
            >
              <i
                class="fas fa-times"
              />
            </button>
          </div>
        </div>
      </div>
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
